### PR TITLE
as per Roman get reader in order from Authorization > X-Adsws-Uid > session

### PR DESCRIPTION
**Pull request by golnazads**

### CODE REVIEW

File: oraclesrv/views.py [hunk 1]
```diff
@@ -22,17 +22,17 @@ def return_response(results, status_code):
...
     :param session:
     :return:
     """
-    if session:
+    if session and token:
         try:
             current_app.logger.info('getting user info from adsws for %s' % (session))
             url = current_app.config['ORACLE_SERVICE_ACCOUNT_INFO_URL'] + '/' + session
-            headers = {'Authorization': 'Bearer ' + current_app.config['ORACLE_SERVICE_ADSWS_API_TOKEN']}
+            headers = {'Authorization': 'Bearer ' + token}
```
Comment (romanchyla) [inline oraclesrv/views.py:35]: nope, this has to stay as it was

File: oraclesrv/views.py [hunk 2]
```diff
@@ -22,17 +22,17 @@ def return_response(results, status_code):
...
 
-def get_user_info_from_adsws(session):
+def get_user_info_from_adsws_having_token(session, token):
     """
 
     :param session:
     :return:
     """
-    if session:
+    if session and token:
```
Comment (romanchyla) [inline oraclesrv/views.py:31]: should be:

if reader:
     url = current_app.config['ORACLE_SERVICE_ACCOUNT_INFO_URL'] + '/' + reader
elif token:
     url = current_app.config['ORACLE_SERVICE_ACCOUNT_INFO_URL'] + '/' + token
elif uid:
     url = current_app.config['ORACLE_SERVICE_ACCOUNT_INFO_URL'] + '/' + uid
elif session:
     ....

File: oraclesrv/views.py [hunk 3]
```diff
@@ -120,8 +138,17 @@ def read_history_get(function, reader):
         the_function = verify_the_function(get_requests_params(payload, 'function', None))
     if the_reader is None:
         the_reader = get_requests_params(payload, 'reader', None)
+        if the_reader is None:
+            session = request.cookies.get('session', None)
+            user_token = request.headers.get('Authorization', None)
```
Comment (romanchyla) [inline oraclesrv/views.py:143]: user_token = request.headers.get('Authorization', '')[7:].strip()

File: oraclesrv/views.py (hunk 2 above)
Comment (golnazads) [reply oraclesrv/views.py:31]: sorry, I don't understand the first line. when we have the reader, we are all set, right, why should we call adsws now?

File: oraclesrv/views.py (hunk 1 above)
Comment (golnazads) [reply oraclesrv/views.py]: done

File: oraclesrv/views.py (hunk 3 above)
Comment (golnazads) [reply oraclesrv/views.py]: done

**Pull request merged**